src/server.c

```diff
@@ -3423,10 +3422,10 @@ void call(client *c, int flags) {
         c->flags |= CLIENT_CLOSE_AFTER_REPLY;
     }
 
-    /* When EVAL is called loading the AOF we don't want commands called
-     * from Lua to go into the slowlog or to populate statistics. */
-    if (server.loading && c->flags & CLIENT_SCRIPT)
-        flags &= ~(CMD_CALL_SLOWLOG | CMD_CALL_STATS);
+    /* When call() is issued during loading the AOF we don't want commands called
+     * from module, exec or LUA to go into the slowlog or to populate statistics. */
+    if ((server.loading && c->flags & CLIENT_SCRIPT) || (c->id == CLIENT_ID_AOF))
```

Comment:
i think we need a two step thing.
1. find the real client, i.e. if this is a script use the server.script_caller, if this is a module, get the real caller too.
2. then check if the real caller is an AOF client.

this practice is done in a few places already, although it looks like not for modules.
it should be impossible to call a script during loading (unless it comes from the AOF), but for modules i think it is possible.
i.e. mark the module command to be allowed during loading, and then use RM_Call.

if it gets complicated (affecting more places), maybe we'll leave that for another PR.

Reply:
ok, so taking modules out of the equation (they're not currently handled by the code), let's also remove them from the comment above this block (or turn it into a TODO).

regarding the current check (using `server.loading`), i think a better one is
`if (c->id == CLIENT_ID_AOF || (c->flags & CLIENT_SCRIPT && server.script_caller->id == CLIENT_ID_AOF))`
or get the `caller` like we do for `CLIENT_TRACKING` below this code.

for some reason i thought that `server.current_client` is set by modules, but now i see it isn't.
so how about doing:
```
client *caller = (c->flags & CLIENT_SCRIPT && server.script_caller) ? server.script_caller : c;
if (caller->id == CLIENT_ID_AOF || server.current_client->id == CLIENT_ID_AOF))
```
we don't propagate scripts to AOF, and we don't allow them during loading, so the only way `c` would be a script is if it is called form a module command, and then current_client should reflect either the network caller or the AOF client.

Reply:
As we discussed offline. I see that the server.current_client is preserved during AOF loading, so I changed the check to reflect that.

Reply:
LGTM